lib/readyset/health/healthchecks.rb

```diff
@@ -0,0 +1,41 @@
+module Readyset
+  module Health
+    # Represents healthchecks that are run against ReadySet to determine whether ReadySet is in a
+    # state where it can serve queries.
+    class Healthchecks
+      def initialize(shard:)
+        @shard = shard
+      end
+
+      # Checks if ReadySet is healthy by invoking `SHOW READYSET STATUS` and checking if
+      # ReadySet is connected to the upstream database.
+      #
+      # @return [Boolean] whether ReadySet is healthy
+      def healthy?
+        connection.execute('SHOW READYSET STATUS').any? do |row|
+          row['name'] == 'Database Connection' && row['value'] == 'Connected'
+        end
+      rescue
+        false
+      end
```

Comment:
**_thought:_** At a glance, this seems closer to a parser :eyeglasses: than a :heavy_check_mark: healthcheck. 

Do you think `row` would be considered referencing something "outside" of the `Healthchecks` object?

Reply:
Hmm I'm not sure I follow -- can you expound a bit?

The health is being checked by ensuring that 1) we can successfully connect to ReadySet and run `SHOW READYSET STATUS` and 2) ReadySet is connected to the upstream database as expected